File: Src/java/cqf-fhir/src/test/java/org/cqframework/fhir/utilities/TestIGContext.java
package org.cqframework.fhir.utilities;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import java.net.URI;
import java.net.URISyntaxException;
import java.net.URL;
import org.hl7.fhir.r5.context.IWorkerContext;
import org.junit.jupiter.api.Test;

public class TestIGContext implements IWorkerContext.ILoggingService {

    @Test
    void typesAndValuesIG() throws URISyntaxException {
        URL url = TestIGContext.class.getResource("types-and-values/ig.ini");
        assertNotNull(url);
        URI uri = Uris.parseOrNull(url.toURI().toString());
        assertNotNull(uri);
        String path = uri.getSchemeSpecificPart();
        IGContext igContext = new IGContext(this);
        igContext.initializeFromIni(path);
        assertEquals("fhir.cqf.typesandvalues", igContext.getPackageId());
        assertEquals("http://fhir.org/guides/cqf/typesandvalues", igContext.getCanonicalBase());
    }

    @Override
    public void logMessage(String s) {
        System.out.println(s);
    }

    @Override
    public void logDebugMessage(LogCategory logCategory, String s) {
        System.out.println(String.format("%s: %s", logCategory.toString(), s));
    }

    @Override
    public boolean isDebugLogging() {
        return true;
    }
}

File: Src/java/cqf-fhir/src/test/kotlin/org/cqframework/fhir/utilities/TestIGContext.kt
package org.cqframework.fhir.utilities

import java.net.URISyntaxException
import org.cqframework.fhir.utilities.Uris.parseOrNull
import org.hl7.fhir.r5.context.ILoggingService
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class TestIGContext : ILoggingService {
    @Test
    @Throws(URISyntaxException::class)
    fun typesAndValuesIG() {
        val url = TestIGContext::class.java.getResource("types-and-values/ig.ini")
        Assertions.assertNotNull(url)
        val uri = parseOrNull(url!!.toURI().toString())
        Assertions.assertNotNull(uri)
        val path = uri!!.getSchemeSpecificPart()
        val igContext = IGContext(this)
        igContext.initializeFromIni(path)
        Assertions.assertEquals("fhir.cqf.typesandvalues", igContext.packageId)
        Assertions.assertEquals(
            "http://fhir.org/guides/cqf/typesandvalues",
            igContext.canonicalBase,
        )
    }

    override fun logMessage(s: String?) {
        println(s)
    }

    override fun logDebugMessage(logCategory: ILoggingService.LogCategory, s: String?) {
        println(String.format("%s: %s", logCategory.toString(), s))
    }

    @Deprecated("Deprecated in FHIR core")
    override fun isDebugLogging(): Boolean {
        return true
    }
}
